### PR TITLE
Feature/homomorphic encryption

### PR DESCRIPTION
**Pull request by liayoo**
- Added homomorphic encryption base functions using desilo
- Added test cases for homomorphic encryption

Example usage:
```
const ain = new Ain("http://node.ainetwork.ai:8080");
await ain.he.init();
const cipherText = ain.he.encrypt(TEST_DATA);
ain.he.TEST_evaluate_double(cipherText); // cipherText updated
const decrypted = ain.he.decrypt(cipherText.save());
```

### CODE REVIEW

File: src/he/desilo.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+}
+
+/**
+ * Factory method for DesiloSeal class
+   @param {Object} keys - object containing sk, pk, galois-key, relin-key strings
+   @returns {DesiloSeal} seal - Desilo SEAL wrapper class
+ */
+async function DesiloSealFactory(keys, params) {
+    const { polyModulusDegree, coeffModulusArray, scaleBit } = params;
+    console.log(params);
```
Comment (shyun-comcom) [inline src/he/desilo.js:179]: Remove console.log?

File: src/he/index.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export default class HomomorphicEncryption {
+  public seal: any;
+  private _initialized: boolean;
+
+  constructor() {
+    this.seal = null;
+    this._initialized = false;
+  }
+
+  async init(keys?: any) {
```
Comment (shyun-comcom) [inline src/he/index.ts:18]: `params` for init parameter? (use current PARAMS value as default)

File: src/he/desilo.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+     * Decrypt ciphertext
+       @param {string::Ciphertext} cipherStr - ciphertext string
+       @returns {Float64Array}
+     */
+    decrypt(cipherStr) {
+        const uploadedCipherText = this.seal.CipherText();
+        uploadedCipherText.load(this.context, cipherStr);
+        const decryptedPlainText = this.decryptor.decrypt(uploadedCipherText);
+        // TODO: data type
+	      return this.encoder.decode(decryptedPlainText);
```
Comment (shyun-comcom) [inline src/he/desilo.js:168]: check indent

File: src/he/desilo.js (hunk 1 above)
Comment (liayoo) [reply src/he/desilo.js]: Done!

File: src/he/index.ts (hunk 2 above)
Comment (liayoo) [reply src/he/index.ts]: Good idea! Done, thanks.

File: src/he/desilo.js (hunk 3 above)
Comment (liayoo) [reply src/he/desilo.js]: Done!

**Comment by liayoo** [comment]
Addressed all comments, PTAL @shyun-comcom

**Review by shyun-comcom — APPROVED** [review]
LGTM!

**Comment by liayoo** [comment]
Thanks for the prompt review! Now merging & releasing v1.1.8 🚀

**Pull request merged**